File: static/js/commands.js
(function (exports) {
    exports.event = {};
    exports.event.UPDATE = 11000;

    exports.editor = {};
    // Your code goes here
    exports.editor.ADD_PRIMITIVE = 2000;
    exports.editor.SET_BACKDROP = 2001;

    exports.project = {};
    exports.project.CREATE = 3000;
    exports.project.DELETE = 3001;
    exports.project.UPDATE = 3002;
    exports.project.SET_ACTIVE = 3003;
    exports.project.SHARE = 3004;
    exports.project.UNSHARE = 3005;
    exports.project.ATTACH_SCENE = 3006;
    exports.project.FIND = 3007;
    exports.project.FIND_ONE = 3008;
    exports.project.GET_ACTIVE = 3009;

    exports.bubbles = {};
    exports.bubbles.CREATE = 4000;
    exports.bubbles.UPDATE = 4001;
    exports.bubbles.DELETE = 4002;
    exports.bubbles.SET_ACTIVE = 4003;
    exports.bubbles.ADD_FLOWOBJECT = 4004;
    exports.bubbles.DEFINE_STATE = 4005;
    exports.bubbles.REMOVE_STATE = 4006;
    exports.bubbles.GET = 4007;
    exports.bubbles.GET_ALL = 4008;

    exports.states = {};
    exports.states.NEW = 5000;
    exports.states.UPDATE = 5001;
    exports.states.DELETE = 5002;
    exports.states.SET_ACTIVE = 5003;
    exports.states.GET = 5007;
    exports.states.GET_ALL = 5008;
    exports.states.TRANSITION = 5009;
    exports.states.ASSOCIATE_FLOW_OBJECT = 5010;
    exports.states.DISASSOCIATE_FLOW_OBJECT = 5011;

    exports.FlowObject = {};
    exports.FlowObject.CREATE = 6000;
    exports.FlowObject.UPDATE = 6001;
    exports.FlowObject.DELETE = 6002;
    exports.FlowObject.SET_PARENT = 6003;
    exports.FlowObject.GET = 6007;
    exports.FlowObject.SELECT = 6008;
    exports.FlowObject.DESELECT = 6009;
    exports.FlowObject.LOCK = 6010;
    exports.FlowObject.FREEZE = 6011;
    exports.FlowObject.panel = {};
    exports.FlowObject.TRANSFORM_UPDATE = 1;
    exports.FlowObject.PROPERTY_UPDATE = 6102;

    exports.FlowObject.command = {};
    exports.FlowObject.command.NEW = 6100;
    exports.FlowObject.command.TRANSLATE = 6101;
    exports.FlowObject.command.ROTATE = 6102;
    exports.FlowObject.command.SCALE = 6103;
    exports.FlowObject.command.SET_PROPERTY = 6104;

    exports.FlowObject.type = {};
    exports.FlowObject.type.PRIMITIVE_CUBE = 6200;
    exports.FlowObject.type.PRIMITIVE_SPHERE = 6201;
    exports.FlowObject.type.TEXT = 6202;
    exports.FlowObject.type.LINE = 6203;
    exports.FlowObject.type.GRID = 6204;

    exports.FlowMark = {};
    exports.FlowAnchor = {};

    exports.FlowSource = {};
    exports.FlowSource.CREATE = 7000;
    exports.FlowSource.UPDATE = 7001;
    exports.FlowSource.DELETE = 7002;

    exports.clients = {};
    exports.clients.register = 0;

    exports.annotations = {};

    exports.user = {};
    exports.user.CREATE = 10000;
    exports.user.UPDATE = 10001;
    exports.user.DELETE = 10002;
    exports.user.FIND = 10003;
    exports.user.SET_ACTIVE_PROJECT = 10004;
    exports.user.SET_ACTIVE_SCENE = 10005;
    exports.user.SET_ACTIVE_STATE = 10006;
    exports.user.GET_USERS = 8;

    exports.client = {};
    exports.client.REGISTER_CLIENT = 0;
    exports.client.LOGIN = 2;
    exports.client.MAKE_MASTER = 3;
    exports.client.CONNECTION_REESTABLISHED = 4;
    exports.client.NEW_CLIENT = 6;
    exports.client.LOGGING_IN = 7;
    exports.client.CLIENT_DISCONNECT = 10;
    exports.client.GET_CLIENTS = 13;
    exports.client.DELETE_CLIENT = 100;

    exports.TRANSFORM_UPDATE = 1;
    exports.ERROR = 5;
    exports.PING = 9;
    exports.PONG = 11;
    exports.user.NEW_USER = 12;


    exports.UPDATE_EXTENSIONS = 1000;
    exports.UPDATE_MODULES = 1001;
    exports.UPDATE_API = 1002;

    exports.ERROR_NO_CLIENT_FOUND = 0;

    exports.CLIENT_EDITOR = 0;
    exports.CLIENT_WEB = 1;
    exports.CLIENT_RIPPLE = 2;
    exports.CLIENT_HOLOLENS = 3;
    exports.CLIENT_MONITOR = 4;

    exports.STATE_CONNECTED = 0;
    exports.STATE_DISCONNECTED = 1;

    exports.test = function () {
        return 'hello world'
    };

})(typeof exports === 'undefined' ? this['commands'] = {} : module.exports);

File: ObjCreationTest/Assets/scripts/Commands.cs
public class Commands
{
    public class Transform
    {
        public const int UPDATE = -2000;
    }
    public class FlowObject {
        public const int CREATE = 1501;
    }
    public class Value
    {
        public const int UPDATE = -1000;
        public class type
        {
            public const int STRING = -1100;
            public const int COLOR = -1101;
            public const int VECTOR3 = -1102;
            public const int VECTOR4 = -1103;
            public const int INT = -1104;
            public const int FLOAT = -1105;
        }
    }
    public class Droplet
    {
        public const int CREATE = 2000;
        public const int NEW_INSTANCE = 2001;
        public const int UPDATE = 2002;
        public const int DELETE = 2003;
        public const int SET_TYPE = 2004;
        public class types
        {
            public const int PRIMITIVE = 2500;
            public const int PLANE = 2501;
            public const int QUAD = 2502;
            public const int MESH = 2503;
            public const int TEXT = 2504;
            public const int SHAPE = 2505;
            public const int LINE = 2506;
            public class primitive
            {
                public const int PRIMITIVE_CUBE = 2100;
                public const int PRIMITIVE_SPHERE = 2101;
                public const int PRIMITIVE_CAPSULE = 2102;
                public const int PRIMITIVE_CYLINDER = 2103;
            }
        }
    }
    public class Modifier
    {
        public const int APPLY = 8000;
        public const int UPDATE = 8001;
        public const int REMOVE = 8002;
        public const int COMMIT = 8003;
        public class types
        {
            public const int ARRAY = 8100;
            public const int SCALE = 8101;
            public const int TRANSLATE = 8102;
            public const int ROTATE = 8103;
            public const int COLOR = 8104;
            public const int TEXTURE = 8104;
            public const int SHADER = 8104;
        }
    }

    public class Editor
    {
        public const int TOGGLE_BACKDROP = 10000;
    }

    public class Project
    {
        public const int CREATE = 3000;
        public const int DELETE = 3001;
        public const int UPDATE = 3002;
        public const int SET_ACTIVE = 3003;
        public const int SHARE = 3004;
        public const int UNSHARE = 3005;
        public const int ATTACH_BUBBLE = 3006;
        public const int FIND = 3007;
        public const int FIND_ONE = 3008;
        public const int GET_ACTIVE = 3009;
        internal static int MIN = 3000;
        internal static int MAX = 3999;
    }
    public class Bubbles
    {
        public const int CREATE = 4000;
        public const int DELETE = 4001;
        public const int UPDATE = 4002;
        public const int SET_ACTIVE = 4003;
        public const int ADD_DROPLET = 4004;
        public const int DEFINE_STATE = 4005;
        public const int REMOVE_STATE = 4006;
        public const int GET = 4007;
        public const int GET_ALL = 4008;
        public class type
        {
            public const int ENVIRONMENT = 4500;
            public const int MARKER = 4501;
            public const int USER = 4502;
            public const int AMBIENT = 4503;
            public const int ANNOTATION = 4504;
        }
    }
    public class States
    {
        public const int UPDATE = 4002;
        public const int DELETE = 4001;
        public const int SET_ACTIVE = 4003;
        public const int GET = 4007;
        public const int GET_ALL = 4008;
        public const int TRANSITION = 5009;
        public const int ASSOCIATE_FLOW_OBJECT = 5010;
        public const int DISASSOCIATE_FLOW_OBJECt = 5011;
    }

    public class User
    {
        public const int LOGIN = 10002;
    }
    public const int REGISTER_CLIENT = 0;
    public const int LOGIN = 2;
    public const int MAKE_MASTER = 3;
    public const int CONNECTION_REESTABLISHED = 4;
    public const int ERROR = 5;
    public const int NEW_CLIENT = 6;
    public const int LOGGING_IN = 7;
    public const int GET_USERS = 8;
    public const int PING = 9;
    public const int CLIENT_DISCONNECT = 10;
    public const int PONG = 11;
    public const int SET_BACKDROP = 1101;
}
